### PR TITLE
Refactors the gathering of OpenProject plugins

### PR DESCRIPTION
**Pull request by jonasheinrich**
This has no ticket. I thought there was a bug to this topic but I wasn't able to find it. 

Without this refactoring, OpenProject plugins that provide a packager.json need to be in the gem group `:opf_plugins`. 

This refactoring is not done yet:
- It does not work by now. Somehow I should get rid of the `require ..` output in the beginning. Or don't pass the result via stdout but via something else. E.g. an environment variable.
- `bundle` gets called explicitly. That might break some installations (e.g. packager).

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://jonasheinrich-feature-refactor-gathering-of-plugins.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://jonasheinrich-feature-refactor-gathering-of-plugins.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: bin/plugin_info.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+def to_gem_name(plugin_name)
```
Comment (NobodysNightmare) [inline bin/plugin_info.rb:1]: OMG! No awesome copyright header!

**Comment by NobodysNightmare** [comment]
I have no great input on the hard questions of your PR.

However what came to my attention is that you are guessing the gem name of a plugin.

Since you are already going the long way to query the "official" ruby interface for plugins, we could use this interface to determine the plugins gem name.

E.g. in the engine of a plugin you could write:

``` ruby
    register 'openproject-fancy-stuff',
             :author_url => 'http://finn.de',
             :requires_openproject => '>= 4.0.0',
             :settings => settings,
             :gem_name: 'op-fancy-stuff' do
```

To reduce noise you could integrate the "guessing" there, by assuming a useful default gem name. Then a plugin would only need to overwrite that name in case it deviates from the norm.

Your line `gem_name = to_gem_name(plugin.id)` could thereby become `gem_name = plugin.gem_name`.

**Comment by jonasheinrich** [comment]
Maybe [this](https://github.com/opf/openproject/commit/7671d13b508a05db7454b1c013ce78ede7e3c1df) helps us with the problem of the noisy output.

**Comment by jonasheinrich** [comment]
Closing this because it unfortunately won't be finished in the next time.

**Pull request closed without merging**